pilot/pkg/model/jwks_resolver.go

```diff
@@ -435,3 +442,57 @@ func (r *JwksResolver) refresh() {
 func (r *JwksResolver) Close() {
 	closeChan <- true
 }
+
+// Compare two JWKS responses, returning true if there is a difference and false otherwise
+func compareJWKSResponse(oldKeyString string, newKeyString string) (bool, error) {
+	if oldKeyString == newKeyString {
+		return false, nil
+	}
+
+	var oldJWKs map[string]interface{}
+	var newJWKs map[string]interface{}
+	e := json.Unmarshal([]byte(newKeyString), &newJWKs)
+	if e != nil {
+		// If the new key is not parseable as JSON return an error since we will not want to use this key
+		log.Warnf("New JWKs public key JSON is not parseable: %s", newKeyString)
+		return false, e
+	}
+	e = json.Unmarshal([]byte(oldKeyString), &oldJWKs)
+	if e != nil {
+		log.Warnf("Previous JWKs public key JSON is not parseable: %s", oldKeyString)
+		return true, nil
+	}
+
+	// Sort both sets of keys by "kid (key ID)" to be able to directly compare
+	oldKeys, oldKeysExists := oldJWKs["keys"].([]interface{})
+	newKeys, newKeysExists := newJWKs["keys"].([]interface{})
+	if oldKeysExists && newKeysExists {
+		sort.Slice(oldKeys, func(i, j int) bool {
+			key1 := oldKeys[i].(map[string]interface{})
```

Comment:
I think you still need `key1, ok := ...` here because the value could be in a different type other than map[string]interface{}. Same for other places.